### PR TITLE
First pass on Info

### PR DESCRIPTION
**Pull request by Andre601**
## Checks

I have...

- [x] read and understood the [Contributing Guidelines][contributing]
- [x] Updated any nessecary files such as the README.md and/or CHANGELOG.md.

## Type of Pull request

- [ ] **Minor Change**  
  This Pull request doesn't break existing configuration.
- [ ] **Major Change**  
  This Pull request will break existing configuration.
- [ ] **Bug fix**  
  This Pull request will fix a (critical) bug.
- [x] **Documentation**  
  This Pull request only changes documentation (README.md, CHANGELOG.md, etc.)
- [ ] **Other:** `__________` 

## Description

First pass on updating the README with valuable information.

Things left to do:
- [x] Add Workflow structure

Closes #

[contributing]: CONTRIBUTING.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,76 @@
...
+  }
+]
+```
+Here is a quick rundown of the different options:
+
+- `search` is the text that the Action should look for in the Template file. It can be any text you like.
+- `replace` is the value that the `search` value should be replaced with. In our example above would any appearance of `{foo}` be replaced with `Bar`.
+- `eval` is a boolean to set whether the `replace` value should be evaluated by the Action for further manipulation. It essentially allows you to set Javascript code that would be executed when the Action runs (i.e. `new Date().toLocaleString()` would be executed as Javascript code and the result used as replacement for the search string).
+
+You can find an [Example JSON file][example] with some recommended placeholders in this Repository.
```
Comment (PuneetGopinath) [inline README.md]:
```suggestion
You can find a [JSON file][default] which would be the default placeholders in the workflow in this repository.
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,76 @@
-# readme-replacer
-Replace data from template repository with dynamic data
+[example]: https://github.com/Readme-Workflows/readme-replacer/blob/main/src/replacers.json
```
Comment (PuneetGopinath) [inline README.md]:
```suggestion
[default]: https://github.com/Readme-Workflows/readme-replacer/blob/main/src/replacers.json
```

**Review by PuneetGopinath — CHANGES_REQUESTED** [review]
Cool! But some small changes

File: README.md [hunk 3]
```diff
@@ -1,2 +1,76 @@
...
+  }
+]
+```
+Here is a quick rundown of the different options:
+
+- `search` is the text that the Action should look for in the Template file. It can be any text you like.
+- `replace` is the value that the `search` value should be replaced with. In our example above would any appearance of `{foo}` be replaced with `Bar`.
+- `eval` is a boolean to set whether the `replace` value should be evaluated by the Action for further manipulation. It essentially allows you to set Javascript code that would be executed when the Action runs (i.e. `new Date().toLocaleString()` would be executed as Javascript code and the result used as replacement for the search string).
+
+You can find a [JSON file][default] which would be the default placeholders in the workflow in this repository.
```
Comment (PuneetGopinath) [inline README.md:73]:
```suggestion
You can find a [JSON file][default] which would be the default placeholders in the workflow.
```
It's not in this repository, that is default for all users, wherever and whoever uses this action will use this as the default replacers

**Review by PuneetGopinath — CHANGES_REQUESTED** [review]
1 minor mistake I had missed out

File: README.md (hunk 3 above)
Comment (PuneetGopinath) [reply README.md:73]: Yeah, I myself forgot to remove that. Didn't notice.

Comment (Andre601) [reply README.md:73]: It's still located within this repository so it's correct.

Comment (PuneetGopinath) [reply README.md:73]: > It's still located within this repository so it's correct.

I don't know what to reply for this, let's wait for @abhijoshi2k

Comment (Andre601) [reply README.md:73]: Just read the message with your changes and tell me if it makes sense.
It say you can FIND a file... where? It wouldn't mention the location.

**Review by abhijoshi2k — COMMENTED** [review]
@Andre601 Can we provide an explanation on how this repository is linked with the `Readme-Icons` repository? The default replacers for readme icons will be added in #2 .

**Comment by Andre601** [comment]
> @Andre601 Can we provide an explanation on how this repository is linked with the `Readme-Icons` repository? The default replacers for readme icons will be added in #2 .

Not really sure it's worth to mention this...
At most maybe a "See also" section that just links to the other projects

**Review by PuneetGopinath — APPROVED** [review]

**Review by abhijoshi2k — APPROVED** [review]

**Comment by abhijoshi2k** [comment]
@Andre601 Should we merge this? Or is there anything left to be added?
Asking because there's still `in progress` label here.

**Comment by PuneetGopinath** [comment]
> @Andre601 Should we merge this? Or is there anything left to be added?
> Asking because there's still `in progress` label here.

I think there is, because Andre added `Add workflow structure` in the description.

**Comment by Andre601** [comment]
I forgot to update the description. It should now be ready, but I would request you to quickly check the workflow setup for if it would be the right syntax, or if it would require some changes.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,76 @@
...
+
+## Setup
+
+To Set up the Action, first create a new workflow file and populate it with the following values:  
+```yaml
+name: Replace placeholders
+
+on:
+  schedule:
+    - cron: '*/30 * * * *' # Activate every 30 minutes
```
Comment (PuneetGopinath) [inline README.md:17]: @abhijoshi2k Is this cron required, or should it trigger on push?

Comment (abhijoshi2k) [reply README.md:17]: It depends on how user uses it. Keep it cron for now. User can change according to needs

**Review by PuneetGopinath — APPROVED** [review]

**Comment by abhijoshi2k** [comment]
```
env:
	GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
@Andre601 We might need token to commit in repo. Refer [this file](https://github.com/abhijoshi2k/abhijoshi2k/blob/master/.github/workflows/test-replacer.yml)

**Comment by PuneetGopinath** [comment]
> I forgot to update the description. It should now be ready, but I would request you to quickly check the workflow setup for if it would be the right syntax, or if it would require some changes.

@abhijoshi2k Check the syntax yourself, and comment if OK.

**Comment by Andre601** [comment]
> ```
> env:
> 	GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
> ```
> 
> @Andre601 We might need token to commit in repo. Refer [this file](https://github.com/abhijoshi2k/abhijoshi2k/blob/master/.github/workflows/test-replacer.yml)

done

**Pull request merged**